src/shard_window_ratio.h

```diff
@@ -42,7 +42,7 @@ static inline size_t calculateEffectiveK(size_t originalK, double ratio, size_t
   // We should not get here if numShards == 1
   RS_LOG_ASSERT(numShards > 1, "Should not calculate effective K for single shard");
 
-  if (ratio == MAX_SHARD_WINDOW_RATIO) {
+  if (ratio == MAX_SHARD_WINDOW_RATIO || !(RSGlobalConfig.enableUnstableFeatures)) {
```

Comment:
We should test the flag also sooner, upon parsing the query. `...=>[KNN ... ] => {$SHARD_K_RATIO: X}` should return syntax error if `enableUnstableFeatures` is false